### PR TITLE
partial revert of [LF] drop retroactive interface instance #18282

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**

### CODE REVIEW

File: sdk/daml-lf/language/daml-lf.bzl [hunk 1]
```diff
@@ -105,48 +105,37 @@ def lf_versions_aggregate(versions):
...
-LF_VERSIONS = [
-    "2.1",
-] + LF_DEV_VERSIONS
-
-# All LF versions suported by the engine
-ENGINE_LF_VERSIONS = [
-    "2.1",
-    "2.dev",
-]
+# fully supported LF dev versions
```
Comment (paulbrauner-da) [inline sdk/daml-lf/language/daml-lf.bzl:108]: "Fully" here means supported by both the compiler and the engine? Or does it mean "as opposed to versions supported by the dar reader"? 

Would you mind adding a comment making the meaning explicit?

File: sdk/daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/LookupError.scala [hunk 2]
```diff
@@ -120,8 +120,7 @@ object Reference {
     override def pretty: String = s"template without contract key $tyCon."
   }
 
-  /** References an interface implementation of interfaceName for templateName,
-    * if a unique one exists.
+  /** References an non-retroactive interface implementation of interfaceName for templateName,
```
Comment (paulbrauner-da) [inline sdk/daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/LookupError.scala:123]:
```suggestion
  /** References an non-retroactive interface implementation of interfaceName for templateName.
```

**Review by paulbrauner-da — APPROVED** [review]
Thanks!

File: sdk/daml-lf/language/daml-lf.bzl (hunk 1 above)
Comment (remyhaemmerle-da) [reply sdk/daml-lf/language/daml-lf.bzl:108]: Good advice

**Pull request merged**